CMSgov/design-system
# [WNMGDS-351] Remove determine props 'type' logic from ChoiceList component

The ChoiceList component has logic that render `checkbox` fields or `radio` buttons for the user depending on certain props and conditions. This isn't very transparent as this logic is done on the program level. The change requires the explicit parameter of `checkbox` or `radio`. 

It defines component type as a required field, hence an explicit component type is required when using the ChoiceList component.  

### Changed
`<ChoiceList>`
- Change props `type` to a Required field
- Remove the logic that determines props `type` value
- Deprecate props `multiple` field and display deprecated message
- Modify example file to reflect the changes
- Modify test file to reflect the changes

`<Choice>`
- Change props `type` to a Required field
- Remove default props `type` value
- Modify `<MonthPicker>` to call `<Choice>` with argument `type="checkbox"`
- Modify example file to reflect the changes
- Modify test file to reflect the changes

### Deprecated
`multiple` props is deprecated, explicitly specify the `type` props instead.

### How to test
`<ChoiceList>`
- Go to Components "Checkbox & Radio" page - check react `<ChoiceList>` example frame functions correctly
- Check developer tools console for warning and deprecated messages -  to test this, modify ChoiceList.example.jsx to replace line 31 type="checkbox" with 'multiple'.:
```Warning: Failed prop type: The prop `type` is marked as required in `ChoiceList`, but its value is `undefined`.```
```[Deprecated]: Please remove the 'multiple' prop in <ChoiceList>, use type="checkbox" instead. This prop is deprecated and will be removed in a future release.```
- Check for warning message when a single 'radio' button is rendered. To test, modify ChoiceList.example.jsx to remove line 10 so that this test contains one `choices` element and type="radio". Check developer tools console for warning messages:
```[Warning]: Use type="checkbox" for components with only one choice. A single radio button is disallowed because it prevents users from deselecting the field.```

`<Choice>`
- Go to Components "Checkbox & Radio" page - check react `<Choice>` example frame functions correctly
- Go to Components "Month Picker" page - check the example functions correctly
